src/main/java/com/iota/iri/controllers/TipsViewModel.java

```diff
@@ -57,13 +67,25 @@ public void setSolid(Hash tip) {
         return hashes;
     }
 
-    public Hash getRandomSolidTipHash() {
+    public List<Hash> getLatestSolidTips(int count) {
+        List<Hash> result = new ArrayList<>();
+        
+        int i = 0;
+        Iterator<Hash> hashIterator = solidTips.iterator();
+        while (hashIterator.hasNext() && i < count) {
+            result.add(hashIterator.next());
+            i++;
+        }
+
+        return result;
+    }
+
+    public Hash getRandomSolidTipHash() throws Exception {
         synchronized (sync) {
-            int size = solidTips.size();
-            if (size == 0) {
-                return null;
```

Comment:
what if no solid tip "exists" (after population the solidTips is empty)
it will still return `null`?

Reply:
The correct behavior is to return only the genesis, I'll add a unit test to verify this happens